packages/filebrowser/src/listing.ts

```diff
@@ -1189,12 +1196,37 @@ export class DirListing extends Widget {
       proposedAction: 'move'
     });
     let basePath = this._model.path;
+
     let paths = toArray(
       map(selectedNames, name => {
         return PathExt.join(basePath, name);
       })
     );
     this._drag.mimeData.setData(CONTENTS_MIME, paths);
+
+    let services = this.model.manager.services;
+    const selectedItems: Contents.IModel[] = [];
+    Object.keys(this._selection).forEach(itemname => {
+      if (this._selection[itemname]) {
+        const item = find(items, item => item.name === itemname);
+        if (
+          item.type === 'file' &&
+          imageRendererFactory.mimeTypes.indexOf(item.mimetype) !== -1
+        ) {
+          selectedItems.push(item);
+        }
+      }
+    });
+    // We thunk this so we don't try to make a network call
+    // when it's not needed. E.g. just moving files around
+    // in a filebrowser
+    if (selectedItems.length > 0) {
+      let attachmentThunks = paths.map(path => {
+        return () => services.contents.get(path);
+      });
+      this._drag.mimeData.setData(ATTACHMENTS_MIME, attachmentThunks);
+    }
+
```

Comment:
An alternative way to do this, would be to:
- Add a thunk for each selected item that has a non-null `mimetype`, with the mimetype ``` `${item.mimetype};closure=true` ```.
- Have the markdown cells drop handler check the mimetype:
  - If the type is either a supported mimetype, or a supported mimetype with a `;closure=true` postfix, it is marked as acceptable.
  - Only call as thunk when postfix is present.

This would let us avoid the `rendermime` dependency for the `filebrowser` package, and make it easier for other things to re-use the same dragging logic (would only need to register a drop target).

Reply:
@Madhu94 Do you think this is a good idea? If so, I could implement it if you don't have the time, but I'd want your input on it first.

Reply:
This is a much better way than just having thunks in a mime type, thanks !!  Is it fine if I wrap this up by tomorrow ? Please do proceed with implementing it if I don't update by tomorrow.

Reply:
> either a supported mimetype, or a supported mimetype with a ;closure=true postfix,

I didn't get the "supported mimetype" ? I assumed as far as the notebook format was concerned, any mime type should be valid. Or do we want only those attachments that lab can render? (which would be just images I think)

[Pushed a commit that just checks for mime types ending with ``;closure=true`` for the immediate moment]